examples/helloworld/lib/helloworld.pb.ex

```diff
@@ -1,26 +1,88 @@
 defmodule Helloworld.HelloRequest do
   @moduledoc false
-  use Protobuf, protoc_gen_elixir_version: "0.10.0", syntax: :proto3
+  use Protobuf, protoc_gen_elixir_version: "0.11.0", syntax: :proto3
 
   field :name, 1, type: :string
 end
 
+defmodule Helloworld.HelloRequestFrom do
+  @moduledoc false
+  use Protobuf, protoc_gen_elixir_version: "0.11.0", syntax: :proto3
+
+  field :name, 1, type: :string
+  field :from, 2, type: :string
+end
+
 defmodule Helloworld.HelloReply do
   @moduledoc false
-  use Protobuf, protoc_gen_elixir_version: "0.10.0", syntax: :proto3
+  use Protobuf, protoc_gen_elixir_version: "0.11.0", syntax: :proto3
 
   field :message, 1, type: :string
   field :today, 2, type: Google.Protobuf.Timestamp
 end
 
-defmodule Helloworld.Greeter.Service do
+defmodule Helloworld.GetMessageRequest do
   @moduledoc false
-  use GRPC.Service, name: "helloworld.Greeter", protoc_gen_elixir_version: "0.10.0"
+  use Protobuf, protoc_gen_elixir_version: "0.11.0", syntax: :proto3
 
-  rpc :SayHello, Helloworld.HelloRequest, Helloworld.HelloReply
+  field :name, 1, type: :string
 end
 
-defmodule Helloworld.Greeter.Stub do
+defmodule Helloworld.Message do
   @moduledoc false
-  use GRPC.Stub, service: Helloworld.Greeter.Service
+  use Protobuf, protoc_gen_elixir_version: "0.11.0", syntax: :proto3
+
+  field :text, 1, type: :string
 end
+
+defmodule Helloworld.Greeter.Service do
+  @moduledoc false
+  use GRPC.Service, name: "helloworld.Greeter", protoc_gen_elixir_version: "0.11.0"
+
+  rpc(:SayHello, Helloworld.HelloRequest, Helloworld.HelloReply, %{
+    http: %{
```

Comment:
Other extensions other than transcoding would also have their code generated ? I mean something like that https://docs.kalix.io/java/writing-grpc-descriptors-protobuf.html#_service

Reply:
Yes! If the extension exists (i.e already generated in the current project) and the code is generated with https://github.com/drowzy/protobuf_generate, the extension will be loaded during code gen and populated as the 4th parmeter in rpc:

```elixir
# in request
stream = %GRPC.Server.Stream{
  __interface__: %{send_reply: &GRPC.Server.Stream.send_reply/3},
  adapter: GRPC.Server.Adapters.Cowboy,
  codec: GRPC.Codec.JSON,
  compressor: nil,
  endpoint: nil,
  grpc_type: :unary,
  http_method: :post,
  http_transcode: true,
  local: nil,
  method_name: "GetMessage",
  payload: %{pid: #PID<0.1631.0>},
  request_id: "FyGm0_ZowKuZONIAAAQD",
  request_mod: Transcode.GetMessageRequest,
  response_mod: Transcode.Message,
  rpc: {:get_message, {Transcode.GetMessageRequest, false},
   {Transcode.Message, false},
   %{
     http: %{
       type: Google.Api.PbExtension,
       value: %Google.Api.HttpRule{
         __unknown_fields__: [],
         additional_bindings: [],
         body: "",
         pattern: {:get, "/v1/messages/{name}"},
         response_body: "",
         selector: ""
       }
     }
   }},
  server: GRPC.Integration.ServerTest.TranscodeServer,
  service_name: "transcode.Messaging"
}

%{value: http_rule} = GRPC.Server.rpc_options(stream.rpc, :http)
``` 
Or by checking the `service.__rpc_calls__`

The helloworld example was generated using:

```
# priv/protos/google/api/annotations.proto
# priv/protos/google/api/http.proto

$ mix protobuf.generate --include-path=priv/protos, --output-path=./lib google/api/http.proto google/api/annotations.proto

# priv/protos/helloword.proto
$ mix protobuf.generate --include-path=priv/protos --plugins=ProtobufGenerate.Plugins.GRPCWithOptions --output-path=./lib priv/protos/helloworld.proto
```

Reply:
I like a lot of this approach
```
%{value: http_rule} = GRPC.Server.rpc_options(stream.rpc, :http)
```